packages/protocol/contracts/layer1/preconf/impl/PreconfWhitelist.sol

```diff
@@ -172,18 +172,31 @@ contract PreconfWhitelist is EssentialContract, IPreconfWhitelist {
 
     function _addOperator(address _operator, uint8 _operatorChangeDelay) internal {
         require(_operator != address(0), InvalidOperatorAddress());
-        require(operators[_operator].activeSince == 0, OperatorAlreadyExists());
 
-        uint8 _operatorCount = operatorCount;
+        OperatorInfo storage info = operators[_operator];
+
+        // if they're already active, just revert
+        if (info.activeSince != 0) {
+            revert OperatorAlreadyExists();
+        }
+
+        // re-activating someone who was scheduled for removal,
+        // but consolidate was not called.
         uint64 activeSince = epochStartTimestamp(_operatorChangeDelay);
-        operators[_operator] = OperatorInfo({
-            activeSince: activeSince,
-            inactiveSince: 0, // no removal scheduled.
-            index: _operatorCount
-        });
-        operatorMapping[_operatorCount] = _operator;
-        unchecked {
-            operatorCount = _operatorCount + 1;
+        if (info.inactiveSince != 0) {
+            info.activeSince = activeSince;
```

Comment:
```
info.activeSince = activeSince;
info.inactiveSince = 0;
```
these two lines are shared by ` if` and `else`, it can be moved out of the if-ese-statement.